### DISCUSSION TITLE
Shipper User: I want carriers to know what they can do if their message was successful but no appointments are available in the Fetch available appointments response.

### DISCUSSION THREAD

**Discussion by OnWednesdaysWeWearFlannel** [freightapis/ssc#12] [Ideas]
Hello!
 
 Shipper User: I want carriers to know what they can do if their message was successful but no appointments are available in the Fetch available appointments response.
 
"If there are no available appointments for the information passed by the carrier / broker, the list of available appointments in the success response SHOULD be empty."
Based on this verbiage, it would be a success message instead of a failure, so there wouldn't be an error where we could place our message. It also specifies being empty. 
https://freightapis.github.io/ssc/#tag/Appointments/operation/fetch-available-appointments
 
Suggestion: Additional optional field to the fetch available appointments response to provide instructions if no appointments are found. 
Unfortunately based on the standards, there is no spot for comments to be provided when no appointments are available. SSC's guidelines on additional fields are outlined in the Adding Custom properties  > Request and Response Bodies "In summary, adding a new property to the request body or headers SHOULD be avoided if possible as it forces more customization between the clients. Feel free to contact the SSC for additional field suggestions to add to the specification."
https://freightapis.github.io/ssc/#section/Adding-Custom-Properties/Request-and-Response-Bodies
 
If we add this, we will comment here how we add it for consistencies. Our PM will also likely reach out on this too, but wanted to make sure it was recorded in this form in case anyone else similarly needed this functionality.
 
-Thanks - Shipper implementation

**Comment by miller79** [comment]
This suggestion looks good. I'll bring this back to the internal team as well for some discussions.
